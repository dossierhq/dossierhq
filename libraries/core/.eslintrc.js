module.exports = {
  env: {
    browser: true,
    es2021: true,
  },
  extends: [
    'eslint:recommended',
    'plugin:@typescript-eslint/recommended',
    'plugin:import/recommended',
    'plugin:import/typescript',
  ],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaVersion: 12,
    sourceType: 'module',
  },
  plugins: ['@typescript-eslint'],
  reportUnusedDisableDirectives: true,
  rules: {
    '@typescript-eslint/consistent-type-imports': ['warn'],
    '@typescript-eslint/no-parameter-properties': ['error'],
    '@typescript-eslint/no-unused-vars': [
      'warn',
      {
        vars: 'local',
        args: 'after-used',
        ignoreRestSiblings: true,
        varsIgnorePattern: '^_',
        argsIgnorePattern: '^_',
      },
    ],
    //TODO figure out how to deal with imports. jest struggles with .js
    // 'import/extensions': ['error', { js: 'always', jsx: 'always', ts: 'always', tsx: 'always' }],
    // 'import/no-unresolved': ['off'],
  },
};
