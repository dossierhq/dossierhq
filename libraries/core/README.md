# `@dossierhq/core`

The core Dossier library used by clients and server alike, used to interact with schema and entities directly, as well as remotely through a client.

`@dossierhq/core` is part of [Dossier](https://www.dossierhq.dev/), a toolkit for building headless CMSs.

- [Changelog](./CHANGELOG.md)
