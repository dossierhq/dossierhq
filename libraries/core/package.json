{
  "name": "@dossierhq/core",
  "version": "0.3.1",
  "description": "The core Dossier library used by clients and server alike, used to interact with schema and entities directly, as well as remotely through a client.",
  "keywords": [
    "dossier",
    "cms",
    "schema",
    "content"
  ],
  "homepage": "https://dossierhq.dev",
  "type": "module",
  "exports": {
    ".": {
      "import": "./lib/esm/index.js",
      "require": {
        "types": "./lib/esm/index.d.ts",
        "default": "./lib/cjs/index.js"
      }
    }
  },
  "types": "./lib/esm/index.d.ts",
  "license": "MIT",
  "scripts": {
    "build": "npm run prettier:check && heft build --clean && npm run test && post-process-build deno cjs-package-json",
    "build:watch": "heft build --watch",
    "clean": "heft clean",
    "eslint:check": "eslint .",
    "eslint:fix": "eslint --fix .",
    "prettier:check": "prettier --check .",
    "prettier:fix": "prettier --write .",
    "test": "vitest",
    "test:update-snapshot": "vitest --update",
    "test:watch": "vitest --watch"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/dossierhq/dossierhq.git",
    "directory": "libraries/core"
  },
  "dependencies": {
    "tslib": "~2.5.3"
  },
  "devDependencies": {
    "@dossierhq/eslint-config-generic": "workspace:*",
    "@dossierhq/heft-node-rig": "workspace:*",
    "@rushstack/heft": "0.50.6",
    "eslint": "~8.43.0",
    "prettier": "2.8.8",
    "typescript": "~5.1.3",
    "vitest": "~0.32.2"
  }
}
