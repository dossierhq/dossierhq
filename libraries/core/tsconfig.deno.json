{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "module": "ES2020",
    "outDir": "lib/deno",
    "moduleResolution": "Node"
  }
}
