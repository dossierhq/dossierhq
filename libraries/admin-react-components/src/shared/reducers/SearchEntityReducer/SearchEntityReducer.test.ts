import type {
  Connection,
  Edge,
  EntitySamplingPayload,
  ErrorType,
  PublishedEntity,
} from '@dossierhq/core';
import { AdminQueryOrder, ok } from '@dossierhq/core';
import { describe, expect, test } from 'vitest';
import type { SearchEntityState, SearchEntityStateAction } from './SearchEntityReducer.js';
import {
  initializeSearchEntityState,
  reduceSearchEntityState,
  SearchEntityStateActions,
} from './SearchEntityReducer.js';

function createPublishedEntity(id: string): PublishedEntity {
  return {
    id,
    info: {
      name: `Entity ${id}`,
      type: 'TitleOnly',
      authKey: 'none',
      createdAt: new Date('2022-03-19T07:51:25.56Z'),
    },
    fields: { title: `Title ${id}` },
  };
}

function createPublishedEntityConnection(
  ids: string[],
  { hasPreviousPage, hasNextPage }: { hasPreviousPage: boolean; hasNextPage: boolean }
): Connection<Edge<PublishedEntity, ErrorType>> {
  return {
    pageInfo: {
      hasPreviousPage,
      hasNextPage,
      startCursor: `cursor-${ids[0]}`,
      endCursor: `cursor-${ids[ids.length - 1]}`,
    },
    edges: ids.map((id) => ({
      cursor: `cursor-${id}`,
      node: ok(createPublishedEntity(id)),
    })),
  };
}

function createPublishedEntitySamplingPayload(
  ids: string[],
  { seed, totalCount }: { seed: number; totalCount: number }
): EntitySamplingPayload<PublishedEntity> {
  return {
    seed,
    totalCount,
    items: ids.map(createPublishedEntity),
  };
}

function reduceSearchEntityStateActions(
  state: SearchEntityState,
  ...actions: SearchEntityStateAction[]
) {
  let newState = state;
  for (const action of actions) {
    newState = reduceSearchEntityState(newState, action);
  }
  return newState;
}

describe('initializeSearchEntityState', () => {
  test('default', () => {
    expect(initializeSearchEntityState({})).toMatchInlineSnapshot(`
      {
        "connection": undefined,
        "connectionError": undefined,
        "entities": null,
        "entitiesScrollToTopSignal": 0,
        "entitySamples": undefined,
        "entitySamplesError": undefined,
        "loadingState": "",
        "paging": {},
        "query": {
          "order": "name",
        },
        "requestedCount": 25,
        "restrictEntityTypes": [],
        "restrictLinksFrom": null,
        "restrictLinksTo": null,
        "sampling": undefined,
        "text": "",
        "totalCount": null,
      }
    `);
  });

  test('restrict entity types', () => {
    expect(initializeSearchEntityState({ restrictEntityTypes: ['Foo', 'Bar'] }))
      .toMatchInlineSnapshot(`
        {
          "connection": undefined,
          "connectionError": undefined,
          "entities": null,
          "entitiesScrollToTopSignal": 0,
          "entitySamples": undefined,
          "entitySamplesError": undefined,
          "loadingState": "",
          "paging": {},
          "query": {
            "entityTypes": [
              "Foo",
              "Bar",
            ],
            "order": "name",
          },
          "requestedCount": 25,
          "restrictEntityTypes": [
            "Foo",
            "Bar",
          ],
          "restrictLinksFrom": null,
          "restrictLinksTo": null,
          "sampling": undefined,
          "text": "",
          "totalCount": null,
        }
      `);
  });

  test('restrict linksFrom', () => {
    const state = initializeSearchEntityState({ restrictLinksFrom: { id: '123' } });
    expect(state).toMatchSnapshot();
    expect(state.query.linksFrom).toEqual({ id: '123' });
  });

  test('restrict linksTo', () => {
    const state = initializeSearchEntityState({ restrictLinksTo: { id: '123' } });
    expect(state).toMatchSnapshot();
    expect(state.query.linksTo).toEqual({ id: '123' });
  });
});

describe('SearchEntityStateActions.SetQuery', () => {
  test('Resets sampling if setting order', () => {
    const state = reduceSearchEntityState(
      initializeSearchEntityState({
        actions: [new SearchEntityStateActions.SetSampling({}, true)],
      }),
      new SearchEntityStateActions.SetQuery(
        { order: AdminQueryOrder.updatedAt },
        { partial: true, resetPagingIfModifying: true }
      )
    );
    expect(state.query).toEqual({ order: AdminQueryOrder.updatedAt });
    expect(state.paging).toEqual({});
    expect(state.sampling).toBeUndefined();
  });

  test('Resets paging (when configured)', () => {
    const state = reduceSearchEntityState(
      initializeSearchEntityState({
        actions: [
          new SearchEntityStateActions.SetPaging({ after: 'cursor', first: 10 }, 'next-page'),
        ],
      }),
      new SearchEntityStateActions.SetQuery(
        { order: AdminQueryOrder.updatedAt },
        { partial: true, resetPagingIfModifying: true }
      )
    );
    expect(state.query).toEqual({ order: AdminQueryOrder.updatedAt });
    expect(state.paging).toEqual({});
    expect(state.sampling).toBeUndefined();
  });

  test('Does not reset paging when changing bounding box and not sampling', () => {
    const state = reduceSearchEntityState(
      initializeSearchEntityState({
        actions: [new SearchEntityStateActions.SetSampling({ count: 10, seed: 123 }, false)],
      }),
      new SearchEntityStateActions.SetQuery(
        { boundingBox: { minLat: 55.07, maxLat: 56.79, minLng: 11.62, maxLng: 16.25 } },
        { partial: true, resetPagingIfModifying: true }
      )
    );
    expect(state.query).toEqual({
      boundingBox: { minLat: 55.07, maxLat: 56.79, minLng: 11.62, maxLng: 16.25 },
    });
    expect(state.paging).toBeUndefined();
    expect(state.sampling).toEqual({ count: 10, seed: 123 });
  });
});

describe('SearchEntityStateActions.SetPaging', () => {
  test('Empty paging', () => {
    const state = reduceSearchEntityState(
      initializeSearchEntityState({}),
      new SearchEntityStateActions.SetPaging({})
    );
    expect(state.paging).toEqual({});
    expect(state.sampling).toBeUndefined();
  });

  test('Resets sampling', () => {
    const state = reduceSearchEntityState(
      initializeSearchEntityState({
        actions: [new SearchEntityStateActions.SetSampling({}, true)],
      }),
      new SearchEntityStateActions.SetPaging({})
    );
    expect(state.paging).toEqual({});
    expect(state.sampling).toBeUndefined();
  });
});

describe('SearchEntityStateActions.SetSampling', () => {
  test('Empty sampling', () => {
    const state = reduceSearchEntityState(
      initializeSearchEntityState({}),
      new SearchEntityStateActions.SetSampling({}, true)
    );
    expect(state.sampling).toEqual({ seed: expect.any(Number) });
    expect(state.paging).toBeUndefined();
  });

  test('Resets paging', () => {
    const state = reduceSearchEntityState(
      initializeSearchEntityState({
        actions: [new SearchEntityStateActions.SetPaging({ first: 200 })],
      }),
      new SearchEntityStateActions.SetSampling({}, true)
    );
    expect(state.sampling).toEqual({ seed: expect.any(Number) });
    expect(state.paging).toBeUndefined();
  });

  test('Resets order', () => {
    const state = reduceSearchEntityState(
      initializeSearchEntityState({
        actions: [
          new SearchEntityStateActions.SetQuery(
            { order: AdminQueryOrder.updatedAt },
            { partial: true, resetPagingIfModifying: true }
          ),
        ],
      }),
      new SearchEntityStateActions.SetSampling({}, true)
    );
    expect(state.sampling).toEqual({ seed: expect.any(Number) });
    expect(state.query).toEqual({});
  });

  test('Partial count, keeps seed', () => {
    const state = reduceSearchEntityState(
      initializeSearchEntityState({
        actions: [new SearchEntityStateActions.SetSampling({ seed: 123 }, false)],
      }),
      new SearchEntityStateActions.SetSampling({ count: 100 }, true)
    );
    expect(state.sampling).toEqual({ seed: 123, count: 100 });
    expect(state.requestedCount).toBe(100);
  });
});

describe('SearchEntityState scenarios', () => {
  test('Next page -> loading -> loaded', () => {
    const initialState = initializeSearchEntityState({
      actions: [
        new SearchEntityStateActions.SetPaging({ first: 1 }),
        new SearchEntityStateActions.UpdateSearchResult(
          createPublishedEntityConnection(['1'], { hasPreviousPage: false, hasNextPage: true }),
          undefined
        ),
        new SearchEntityStateActions.UpdateTotalCount(2),
      ],
    });
    expect(initialState).toMatchSnapshot('1 - initial');

    const loadingState = reduceSearchEntityStateActions(
      initialState,
      new SearchEntityStateActions.SetPaging({ after: 'cursor-1' }, 'next-page'),
      new SearchEntityStateActions.UpdateSearchResult(undefined, undefined)
    );
    expect(loadingState).toMatchSnapshot('2 - loading');

    const loadedState = reduceSearchEntityState(
      loadingState,
      new SearchEntityStateActions.UpdateSearchResult(
        createPublishedEntityConnection(['2'], { hasPreviousPage: true, hasNextPage: false }),
        undefined
      )
    );
    expect(loadedState).toMatchSnapshot('3 - loaded');

    expect(initialState.entitiesScrollToTopSignal).toBeLessThan(
      loadedState.entitiesScrollToTopSignal
    );
  });

  test('Sampling -> change seed -> loading -> loaded', () => {
    const initialState = initializeSearchEntityState({
      actions: [
        new SearchEntityStateActions.SetSampling({ count: 1, seed: 123 }, false),
        new SearchEntityStateActions.UpdateSampleResult(
          createPublishedEntitySamplingPayload(['1'], { seed: 123, totalCount: 2 }),
          undefined
        ),
      ],
    });
    expect(initialState).toMatchSnapshot('1 - initial');

    const loadingState = reduceSearchEntityStateActions(
      initialState,
      new SearchEntityStateActions.SetSampling({ seed: 456 }, true),
      new SearchEntityStateActions.UpdateSampleResult(undefined, undefined)
    );
    expect(loadingState).toMatchSnapshot('2 - loading');

    const loadedState = reduceSearchEntityStateActions(
      loadingState,
      new SearchEntityStateActions.UpdateSampleResult(
        createPublishedEntitySamplingPayload(['2'], { seed: 456, totalCount: 2 }),
        undefined
      )
    );
    expect(loadedState).toMatchSnapshot('3 - loaded');

    expect(initialState.entitiesScrollToTopSignal).toBeLessThan(
      loadedState.entitiesScrollToTopSignal
    );
  });

  test('From searching -> set sampling -> loading -> loaded', () => {
    const initialState = initializeSearchEntityState({
      actions: [
        new SearchEntityStateActions.SetPaging({ first: 1 }),
        new SearchEntityStateActions.UpdateSearchResult(
          createPublishedEntityConnection(['1'], { hasPreviousPage: false, hasNextPage: true }),
          undefined
        ),
        new SearchEntityStateActions.UpdateTotalCount(2),
        new SearchEntityStateActions.UpdateSampleResult(undefined, undefined),
      ],
    });
    expect(initialState).toMatchSnapshot('1 - initial');

    const loadingState = reduceSearchEntityStateActions(
      initialState,
      new SearchEntityStateActions.SetSampling({ count: 1, seed: 123 }, false),
      new SearchEntityStateActions.UpdateSearchResult(undefined, undefined),
      new SearchEntityStateActions.UpdateTotalCount(null)
    );
    expect(loadingState).toMatchSnapshot('2 - loading');

    const loadedState = reduceSearchEntityState(
      loadingState,
      new SearchEntityStateActions.UpdateSampleResult(
        createPublishedEntitySamplingPayload(['2'], { seed: 123, totalCount: 1 }),
        undefined
      )
    );
    expect(loadedState).toMatchSnapshot('3 - loaded');

    expect(initialState.entitiesScrollToTopSignal).toBeLessThan(
      loadedState.entitiesScrollToTopSignal
    );
  });

  test('From sampling -> set paging -> loading -> loaded', () => {
    const initialState = initializeSearchEntityState({
      actions: [
        new SearchEntityStateActions.SetSampling({ count: 1, seed: 123 }, false),
        new SearchEntityStateActions.UpdateSampleResult(
          createPublishedEntitySamplingPayload(['1'], { seed: 123, totalCount: 1 }),
          undefined
        ),
      ],
    });
    expect(initialState).toMatchSnapshot('1 - initial');

    const loadingState = reduceSearchEntityStateActions(
      initialState,
      new SearchEntityStateActions.SetPaging({ first: 1 }, 'first-page'),
      new SearchEntityStateActions.UpdateSampleResult(undefined, undefined),
      new SearchEntityStateActions.UpdateSearchResult(undefined, undefined),
      new SearchEntityStateActions.UpdateTotalCount(null)
    );
    expect(loadingState).toMatchSnapshot('2 - loading');

    const loadedState = reduceSearchEntityStateActions(
      loadingState,
      new SearchEntityStateActions.UpdateSearchResult(
        createPublishedEntityConnection(['2'], { hasPreviousPage: false, hasNextPage: true }),
        undefined
      ),
      new SearchEntityStateActions.UpdateTotalCount(2)
    );
    expect(loadedState).toMatchSnapshot('3 - loaded');

    expect(initialState.entitiesScrollToTopSignal).toBeLessThan(
      loadedState.entitiesScrollToTopSignal
    );
  });

  test('Refresh', () => {
    const initialState = initializeSearchEntityState({
      actions: [
        new SearchEntityStateActions.SetPaging({ first: 1 }),
        new SearchEntityStateActions.UpdateSearchResult(
          createPublishedEntityConnection(['1'], { hasPreviousPage: false, hasNextPage: true }),
          undefined
        ),
        new SearchEntityStateActions.UpdateTotalCount(2),
      ],
    });
    expect(initialState).toMatchSnapshot('1 - initial');

    const loadedState = reduceSearchEntityStateActions(
      initialState,
      new SearchEntityStateActions.UpdateSearchResult(
        createPublishedEntityConnection(['2'], { hasPreviousPage: false, hasNextPage: true }),
        undefined
      )
    );
    expect(loadedState).toMatchSnapshot('2 - loaded');

    expect(initialState.entitiesScrollToTopSignal).toEqual(loadedState.entitiesScrollToTopSignal);
  });
});
