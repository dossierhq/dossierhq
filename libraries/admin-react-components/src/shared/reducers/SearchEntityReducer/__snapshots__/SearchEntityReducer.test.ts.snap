// Vitest Snapshot v1

exports[`SearchEntityState scenarios > From sampling -> set paging -> loading -> loaded > 1 - initial 1`] = `
{
  "connection": undefined,
  "connectionError": undefined,
  "entities": [
    OkResult {
      "value": {
        "fields": {
          "title": "Title 1",
        },
        "id": "1",
        "info": {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 1",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 1,
  "entitySamples": {
    "items": [
      {
        "fields": {
          "title": "Title 1",
        },
        "id": "1",
        "info": {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 1",
          "type": "TitleOnly",
        },
      },
    ],
    "seed": 123,
    "totalCount": 1,
  },
  "entitySamplesError": undefined,
  "loadingState": "",
  "paging": undefined,
  "query": {},
  "requestedCount": 1,
  "restrictEntityTypes": [],
  "sampling": {
    "count": 1,
    "seed": 123,
  },
  "text": "",
  "totalCount": null,
}
`;

exports[`SearchEntityState scenarios > From sampling -> set paging -> loading -> loaded > 2 - loading 1`] = `
{
  "connection": undefined,
  "connectionError": undefined,
  "entities": [
    OkResult {
      "value": {
        "fields": {
          "title": "Title 1",
        },
        "id": "1",
        "info": {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 1",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 1,
  "entitySamples": undefined,
  "entitySamplesError": undefined,
  "loadingState": "first-page",
  "paging": {
    "first": 1,
  },
  "query": {},
  "requestedCount": 1,
  "restrictEntityTypes": [],
  "sampling": undefined,
  "text": "",
  "totalCount": null,
}
`;

exports[`SearchEntityState scenarios > From sampling -> set paging -> loading -> loaded > 3 - loaded 1`] = `
{
  "connection": {
    "edges": [
      {
        "cursor": "cursor-2",
        "node": OkResult {
          "value": {
            "fields": {
              "title": "Title 2",
            },
            "id": "2",
            "info": {
              "authKey": "none",
              "createdAt": "2022-03-19T07:51:25.56Z",
              "name": "Entity 2",
              "type": "TitleOnly",
            },
          },
        },
      },
    ],
    "pageInfo": {
      "endCursor": "cursor-2",
      "hasNextPage": true,
      "hasPreviousPage": false,
      "startCursor": "cursor-2",
    },
  },
  "connectionError": undefined,
  "entities": [
    OkResult {
      "value": {
        "fields": {
          "title": "Title 2",
        },
        "id": "2",
        "info": {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 2",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 2,
  "entitySamples": undefined,
  "entitySamplesError": undefined,
  "loadingState": "",
  "paging": {
    "first": 1,
  },
  "query": {},
  "requestedCount": 1,
  "restrictEntityTypes": [],
  "sampling": undefined,
  "text": "",
  "totalCount": 2,
}
`;

exports[`SearchEntityState scenarios > From searching -> set sampling -> loading -> loaded > 1 - initial 1`] = `
{
  "connection": {
    "edges": [
      {
        "cursor": "cursor-1",
        "node": OkResult {
          "value": {
            "fields": {
              "title": "Title 1",
            },
            "id": "1",
            "info": {
              "authKey": "none",
              "createdAt": "2022-03-19T07:51:25.56Z",
              "name": "Entity 1",
              "type": "TitleOnly",
            },
          },
        },
      },
    ],
    "pageInfo": {
      "endCursor": "cursor-1",
      "hasNextPage": true,
      "hasPreviousPage": false,
      "startCursor": "cursor-1",
    },
  },
  "connectionError": undefined,
  "entities": [
    OkResult {
      "value": {
        "fields": {
          "title": "Title 1",
        },
        "id": "1",
        "info": {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 1",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 0,
  "entitySamples": undefined,
  "entitySamplesError": undefined,
  "loadingState": "",
  "paging": {
    "first": 1,
  },
  "query": {
    "order": "name",
  },
  "requestedCount": 1,
  "restrictEntityTypes": [],
  "sampling": undefined,
  "text": "",
  "totalCount": 2,
}
`;

exports[`SearchEntityState scenarios > From searching -> set sampling -> loading -> loaded > 2 - loading 1`] = `
{
  "connection": undefined,
  "connectionError": undefined,
  "entities": [
    OkResult {
      "value": {
        "fields": {
          "title": "Title 1",
        },
        "id": "1",
        "info": {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 1",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 0,
  "entitySamples": undefined,
  "entitySamplesError": undefined,
  "loadingState": "sample",
  "paging": undefined,
  "query": {},
  "requestedCount": 1,
  "restrictEntityTypes": [],
  "sampling": {
    "count": 1,
    "seed": 123,
  },
  "text": "",
  "totalCount": null,
}
`;

exports[`SearchEntityState scenarios > From searching -> set sampling -> loading -> loaded > 3 - loaded 1`] = `
{
  "connection": undefined,
  "connectionError": undefined,
  "entities": [
    OkResult {
      "value": {
        "fields": {
          "title": "Title 2",
        },
        "id": "2",
        "info": {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 2",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 1,
  "entitySamples": {
    "items": [
      {
        "fields": {
          "title": "Title 2",
        },
        "id": "2",
        "info": {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 2",
          "type": "TitleOnly",
        },
      },
    ],
    "seed": 123,
    "totalCount": 1,
  },
  "entitySamplesError": undefined,
  "loadingState": "",
  "paging": undefined,
  "query": {},
  "requestedCount": 1,
  "restrictEntityTypes": [],
  "sampling": {
    "count": 1,
    "seed": 123,
  },
  "text": "",
  "totalCount": null,
}
`;

exports[`SearchEntityState scenarios > Next page -> loading -> loaded > 1 - initial 1`] = `
{
  "connection": {
    "edges": [
      {
        "cursor": "cursor-1",
        "node": OkResult {
          "value": {
            "fields": {
              "title": "Title 1",
            },
            "id": "1",
            "info": {
              "authKey": "none",
              "createdAt": "2022-03-19T07:51:25.56Z",
              "name": "Entity 1",
              "type": "TitleOnly",
            },
          },
        },
      },
    ],
    "pageInfo": {
      "endCursor": "cursor-1",
      "hasNextPage": true,
      "hasPreviousPage": false,
      "startCursor": "cursor-1",
    },
  },
  "connectionError": undefined,
  "entities": [
    OkResult {
      "value": {
        "fields": {
          "title": "Title 1",
        },
        "id": "1",
        "info": {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 1",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 0,
  "entitySamples": undefined,
  "entitySamplesError": undefined,
  "loadingState": "",
  "paging": {
    "first": 1,
  },
  "query": {
    "order": "name",
  },
  "requestedCount": 1,
  "restrictEntityTypes": [],
  "sampling": undefined,
  "text": "",
  "totalCount": 2,
}
`;

exports[`SearchEntityState scenarios > Next page -> loading -> loaded > 2 - loading 1`] = `
{
  "connection": undefined,
  "connectionError": undefined,
  "entities": [
    OkResult {
      "value": {
        "fields": {
          "title": "Title 1",
        },
        "id": "1",
        "info": {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 1",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 0,
  "entitySamples": undefined,
  "entitySamplesError": undefined,
  "loadingState": "next-page",
  "paging": {
    "after": "cursor-1",
  },
  "query": {
    "order": "name",
  },
  "requestedCount": 1,
  "restrictEntityTypes": [],
  "sampling": undefined,
  "text": "",
  "totalCount": 2,
}
`;

exports[`SearchEntityState scenarios > Next page -> loading -> loaded > 3 - loaded 1`] = `
{
  "connection": {
    "edges": [
      {
        "cursor": "cursor-2",
        "node": OkResult {
          "value": {
            "fields": {
              "title": "Title 2",
            },
            "id": "2",
            "info": {
              "authKey": "none",
              "createdAt": "2022-03-19T07:51:25.56Z",
              "name": "Entity 2",
              "type": "TitleOnly",
            },
          },
        },
      },
    ],
    "pageInfo": {
      "endCursor": "cursor-2",
      "hasNextPage": false,
      "hasPreviousPage": true,
      "startCursor": "cursor-2",
    },
  },
  "connectionError": undefined,
  "entities": [
    OkResult {
      "value": {
        "fields": {
          "title": "Title 2",
        },
        "id": "2",
        "info": {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 2",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 1,
  "entitySamples": undefined,
  "entitySamplesError": undefined,
  "loadingState": "",
  "paging": {
    "after": "cursor-1",
  },
  "query": {
    "order": "name",
  },
  "requestedCount": 1,
  "restrictEntityTypes": [],
  "sampling": undefined,
  "text": "",
  "totalCount": 2,
}
`;

exports[`SearchEntityState scenarios > Refresh > 1 - initial 1`] = `
{
  "connection": {
    "edges": [
      {
        "cursor": "cursor-1",
        "node": OkResult {
          "value": {
            "fields": {
              "title": "Title 1",
            },
            "id": "1",
            "info": {
              "authKey": "none",
              "createdAt": "2022-03-19T07:51:25.56Z",
              "name": "Entity 1",
              "type": "TitleOnly",
            },
          },
        },
      },
    ],
    "pageInfo": {
      "endCursor": "cursor-1",
      "hasNextPage": true,
      "hasPreviousPage": false,
      "startCursor": "cursor-1",
    },
  },
  "connectionError": undefined,
  "entities": [
    OkResult {
      "value": {
        "fields": {
          "title": "Title 1",
        },
        "id": "1",
        "info": {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 1",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 0,
  "entitySamples": undefined,
  "entitySamplesError": undefined,
  "loadingState": "",
  "paging": {
    "first": 1,
  },
  "query": {
    "order": "name",
  },
  "requestedCount": 1,
  "restrictEntityTypes": [],
  "sampling": undefined,
  "text": "",
  "totalCount": 2,
}
`;

exports[`SearchEntityState scenarios > Refresh > 2 - loaded 1`] = `
{
  "connection": {
    "edges": [
      {
        "cursor": "cursor-2",
        "node": OkResult {
          "value": {
            "fields": {
              "title": "Title 2",
            },
            "id": "2",
            "info": {
              "authKey": "none",
              "createdAt": "2022-03-19T07:51:25.56Z",
              "name": "Entity 2",
              "type": "TitleOnly",
            },
          },
        },
      },
    ],
    "pageInfo": {
      "endCursor": "cursor-2",
      "hasNextPage": true,
      "hasPreviousPage": false,
      "startCursor": "cursor-2",
    },
  },
  "connectionError": undefined,
  "entities": [
    OkResult {
      "value": {
        "fields": {
          "title": "Title 2",
        },
        "id": "2",
        "info": {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 2",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 0,
  "entitySamples": undefined,
  "entitySamplesError": undefined,
  "loadingState": "",
  "paging": {
    "first": 1,
  },
  "query": {
    "order": "name",
  },
  "requestedCount": 1,
  "restrictEntityTypes": [],
  "sampling": undefined,
  "text": "",
  "totalCount": 2,
}
`;

exports[`SearchEntityState scenarios > Sampling -> change seed -> loading -> loaded > 1 - initial 1`] = `
{
  "connection": undefined,
  "connectionError": undefined,
  "entities": [
    OkResult {
      "value": {
        "fields": {
          "title": "Title 1",
        },
        "id": "1",
        "info": {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 1",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 1,
  "entitySamples": {
    "items": [
      {
        "fields": {
          "title": "Title 1",
        },
        "id": "1",
        "info": {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 1",
          "type": "TitleOnly",
        },
      },
    ],
    "seed": 123,
    "totalCount": 2,
  },
  "entitySamplesError": undefined,
  "loadingState": "",
  "paging": undefined,
  "query": {},
  "requestedCount": 1,
  "restrictEntityTypes": [],
  "sampling": {
    "count": 1,
    "seed": 123,
  },
  "text": "",
  "totalCount": null,
}
`;

exports[`SearchEntityState scenarios > Sampling -> change seed -> loading -> loaded > 2 - loading 1`] = `
{
  "connection": undefined,
  "connectionError": undefined,
  "entities": [
    OkResult {
      "value": {
        "fields": {
          "title": "Title 1",
        },
        "id": "1",
        "info": {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 1",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 1,
  "entitySamples": undefined,
  "entitySamplesError": undefined,
  "loadingState": "sample",
  "paging": undefined,
  "query": {},
  "requestedCount": 1,
  "restrictEntityTypes": [],
  "sampling": {
    "count": 1,
    "seed": 456,
  },
  "text": "",
  "totalCount": null,
}
`;

exports[`SearchEntityState scenarios > Sampling -> change seed -> loading -> loaded > 3 - loaded 1`] = `
{
  "connection": undefined,
  "connectionError": undefined,
  "entities": [
    OkResult {
      "value": {
        "fields": {
          "title": "Title 2",
        },
        "id": "2",
        "info": {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 2",
          "type": "TitleOnly",
        },
      },
    },
  ],
  "entitiesScrollToTopSignal": 2,
  "entitySamples": {
    "items": [
      {
        "fields": {
          "title": "Title 2",
        },
        "id": "2",
        "info": {
          "authKey": "none",
          "createdAt": "2022-03-19T07:51:25.56Z",
          "name": "Entity 2",
          "type": "TitleOnly",
        },
      },
    ],
    "seed": 456,
    "totalCount": 2,
  },
  "entitySamplesError": undefined,
  "loadingState": "",
  "paging": undefined,
  "query": {},
  "requestedCount": 1,
  "restrictEntityTypes": [],
  "sampling": {
    "count": 1,
    "seed": 456,
  },
  "text": "",
  "totalCount": null,
}
`;
